sky/cloud_stores.py

```diff
@@ -114,7 +114,11 @@ class GcsCloudStorage(CloudStorage):
     def _gsutil_command(self):
         gsutil_alias, alias_gen = data_utils.get_gsutil_command()
         return (f'{alias_gen}; GOOGLE_APPLICATION_CREDENTIALS='
-                f'{gcp.DEFAULT_GCP_APPLICATION_CREDENTIAL_PATH} {gsutil_alias}')
+                f'{gcp.DEFAULT_GCP_APPLICATION_CREDENTIAL_PATH}; '
+                'gcloud auth activate-service-account '
```

Comment:
Does something similar need to be done for our VM provisioning pipeline on the jobs/serve controller?

For example, if the user is using service account for auth, runs a controller on k8s and wants to launch a managed job on GCP, will he run into the same issue (see #4512)?

Reply:
Good point! I changed the PR to fix the issue.